src/quemb/shared/manage_scratch.py

```diff
@@ -65,7 +105,7 @@ def __exit__(
         value: BaseException | None,
         traceback: TracebackType | None,
     ) -> Literal[False]:
-        if value is None:
+        if self.cleanup_at_end:
```

Comment:
this change might be relevant for you @lweisburn .

If the WorkDir is used as a contextmanager, then the files are deleted upon exiting the with-statement, even if errors occur.

This is different to the behavior, when calling it in the normal way, where the `atexit.register` means that the WorkDir is only cleaned, if the calculation exits cleanly.

Since you can now pass a WorkDir to BE and it is passed all the way down, this means you can have a script and multiple BE calculations that are guaranteed to be cleaned up, even if errors occur if you write:

```python3
with WorkDir() as scratch:
    BE(..., scratch_dir=scratch)
# here scratch is deleted, even if errors happen.
```
while the normal way of writing it:
```python3
BE(...)
```
cleans the files only upon a clean exit.

Reply:
How cool!! This will be incredibly helpful, thank you!